jerry-core/ecma/base/ecma-helpers-value.c

```diff
@@ -442,22 +618,173 @@ ecma_copy_value_if_not_object (ecma_value_t value) /**< value description */
 } /* ecma_copy_value_if_not_object */
 
 /**
+ * Assign a new value to an ecma-value
+ *
+ * Note:
+ *      value previously stored in the property is freed
+ */
+void
+ecma_value_assign_value (ecma_value_t *value_p, /**< [in, out] ecma value */
+                         ecma_value_t ecma_value) /**< value to assign */
+{
+  JERRY_ASSERT (ECMA_TYPE_NON_PTR_VALUE == 0);
```

Comment:
What does this assert check? Both operands of the equality check seem to be constant.